src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -148,13 +129,19 @@ public interface IChainedHeaderTree
         /// <param name="block">The block.</param>
         /// <returns>Newly created and connected chained header for the specified block.</returns>
         ChainedHeader CreateChainedHeaderWithBlock(Block block);
+
+        /// <summary>
+        /// Get the block and its chained header if it exists.
+        /// </summary>
+        /// <returns>The block and its chained header.</returns>
+        ChainedHeaderBlock GetChainedHeaderBlock(uint256 blockHash);
```

Comment:
We should specify when the return value is null and also when the block part of the result is null.